### PR TITLE
Restructure everything

### PR DESCRIPTION
**Pull request by NicklasHugoy**
Extract creation of JSON, data processing and filters into separate classes

### CODE REVIEW

File: src/data_handler.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from src.smiley_extractor import SmileyExtractor
+from src.data_processor import DataProcessor
+
+
+class DataHandler:
+    def __init__(self, *args, **kwargs) -> None:
+        sample_size = kwargs.pop('sample', 0)
+        skip_scrape = kwargs.pop('no_scrape', False)
+        outputter = get_outputter(kwargs.pop('push', False))
+        self.dataProcessor = DataProcessor(sample_size, skip_scrape, outputter)
```
Comment (Badgie) [inline src/data_handler.py]: _

File: src/smiley_extractor.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        for row in list(root):
+            new_obj = {col.tag: col.text for col in row}
+            SmileyExtractor._convert_to_float(new_obj, 'Geo_Lat', 'Geo_Lng')
+            SmileyExtractor._convert_to_int(new_obj, 'seneste_kontrol', 'naestseneste_kontrol',
+                                            'tredjeseneste_kontrol', 'fjerdeseneste_kontrol')
+            SmileyExtractor._strip_whitespace(new_obj, 'navn1')
+            res.append(new_obj)
+
+        with open(SmileyExtractor.SMILEY_JSON, 'w') as f:
+            f.write(json.dumps(res, indent=4, sort_keys=True))
```
Comment (Badgie) [inline src/smiley_extractor.py]: Why is this static? Induces unnecessary calls through the class rather than the instance

Comment (Badgie) [reply src/smiley_extractor.py]: If you want them to be callable without creating an instance, make this a class method.

File: src/data_processor.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+from src.data_outputter import BaseDataOutputter
+import json
+import time
+from src.temp_file import TempFile
+from src.prev_processed_file import PrevProcessedFile
+from src.cvr import get_cvr_handler, FindSmileyHandler
+from src.filters import Filters
+
+
+class DataProcessor():
```
Comment (Badgie) [inline src/data_processor.py]: No need for `()`

File: src/smiley_extractor.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                data[k]) == str else data[k]
+
+    @staticmethod
+    def _retrieve_smiley_data() -> None:
+        """
+        Download smiley XML data from Fødevarestyrelsen.
+        """
+        res = get(SmileyExtractor.SMILEY_XML_URL)
+        with open(SmileyExtractor.SMILEY_XML, 'w') as f:
+            f.write(res.content.decode('utf-8'))
```
Comment (Badgie) [inline src/smiley_extractor.py]: Same comment as the other method in this file

**Review by Badgie — CHANGES_REQUESTED** [review]
Few comments and some other things.

---

This is not C# >: ⁽
Rename the dir to `filter_xml`.

---

In `data_outputter.py` (could not comment as there are no changes):
While this PR does not change anything here, there are some linting errors and type hints that need fixing.
- 2 newlines between classes/functions
- 1 newline between methods
- The `write` method of the base class has a param with a different name from the inherited classes. These should all be the same.
    - The `write` method expects a `dict`, but most (if not all) of the time we send a `list`. I'd suggest changing the type hint to `Union[dict, list]`, where union is from the `typing` module.
- `should_send_to_db` in `get_outputter` needs a type hint

---

Since we are changing this to a package, you should make an `__init__.py` file in the package dir:
- define the argparser and its arguments globally
- make a `run()` function that parses the args and runs collection
Then you can change `run.py` (of course while adhering to the suggested directory name change) to
```py
from src import run

if __name__ == '__main__':
    run()
```

File: src/data_handler.py (hunk 1 above)
Comment (NicklasHugoy) [reply src/data_handler.py]: What's the problem?

Comment (Badgie) [reply src/data_handler.py]: `self.dataProcessor`

**Review by Badgie — APPROVED** [review]

**Pull request merged**